### ISSUE TITLE
Git LFS not being pushed to remote

### ISSUE THREAD

**Issue by jakkrobbit** [git-lfs/git-lfs#3051]
I've been battling this issue for the past 2 days. LFS appears to be tracking changes to files in my repository, but when I try to push changes, I get an error message of some kind.

At first I had the remote repo on a cloud storage account (Mega), and each time I tried to push a commit I would get something similar to the following error:
```
Remote "origin" does not support the LFS locking API. Consider disabling it with:
  $ git config 'http://fmrepo:8080/comic.git/info/lfs.locksverify' false
Git LFS: (0 of 1 files) 0 B / 54.17 MB
error: failed to push some refs to 'http://fmrepo:8080/comic.git'
```
I thought it might have something to do with the fact I wasn't pushing to an actual server. So, I switched to gitlab, but I'm still having trouble. I edited a psd file to test it, and this is what `git status` and `git lfs status` return:
```
git status
On branch master
Your branch is ahead of 'origin/master' by 1 commit.
  (use "git push" to publish your local commits)
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   PSDs/FM_ch1-pg5.psd

no changes added to commit (use "git add" and/or "git commit -a")
```
```
git lfs status
On branch master
Git LFS objects to be pushed to origin/master:

        PSDs/FM_ch1-pg5.psd (%!s(MISSING))

Git LFS objects to be committed:


Git LFS objects not staged for commit:

        PSDs/FM_ch1-pg5.psd (LFS: 73cee30 -> File: ddb78eb)
```

So, it appears to tracking files correctly (although, I don't know what the "%!s(MISSING)" thing is - that's new as of writing this), but when I try to commit and push changes, I get the following error.

```
Remote "origin" does not support the LFS locking API. Consider disabling it with:
  $ git config 'lfs.https://gitlab.com/findmercy/ch1.git/info/lfs.locksverify' false
Git LFS: (0 of 1 files) 0 B / 54.17 MB
batch request: Permission denied (publickey).: exit status 255
error: failed to push some refs to 'git@gitlab.com:findmercy/ch1.git'
```

I assume this has something to so with the ssh key, but I'm stumped as to how to fix it. I've seen several issues similar to mine (e.g. #2635, #1068, #1726), but the solutions didn't work.

Here's the environment. I'm just using Windows 10's command prompt.
```
git lfs env
git-lfs/2.1.1 (GitHub; windows amd64; go 1.8.1; git 3314e287)
git version 2.13.2.windows.1

Endpoint=https://gitlab.com/findmercy/ch1.git/info/lfs (auth=none)
  SSH=git@gitlab.com:findmercy/ch1.git
LocalWorkingDir=C:\Users\Nikki\Pictures\Drawings & Designs\Find Mercy\Comic\Ch. 1
LocalGitDir=C:\Users\Nikki\Pictures\Drawings & Designs\Find Mercy\Comic\Ch. 1\.git
LocalGitStorageDir=C:\Users\Nikki\Pictures\Drawings & Designs\Find Mercy\Comic\Ch. 1\.git
LocalMediaDir=C:\Users\Nikki\Pictures\Drawings & Designs\Find Mercy\Comic\Ch. 1\.git\lfs\objects
LocalReferenceDir=
TempDir=C:\Users\Nikki\Pictures\Drawings & Designs\Find Mercy\Comic\Ch. 1\.git\lfs\tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
GIT_LFS_PATH=C:\Program Files\Git LFS
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Comment by jomark-t** [comment]
TL;DR did you try `git lfs push`?

**Comment by jakkrobbit** [comment]
I did, but it gave me the same `batch request: Permission denied (publickey).: exit status 255` error.

**Comment by jomark-t** [comment]
Did you try this command `git config 'http://fmrepo:8080/comic.git/info/lfs.locksverify' false`

**Comment by jakkrobbit** [comment]
I tried that, too. It gives me `error: invalid key: 'lfs.https://gitlab.com/findmercy/ch1.git/info/lfs.locksverify'`

**Comment by ttaylorr** [comment]
> ```ShellSession
> $ git lfs status
> On branch master
> Git LFS objects to be pushed to origin/master:
> 
>         PSDs/FM_ch1-pg5.psd (%!s(MISSING))
> 
> Git LFS objects to be committed:
> 
> 
> Git LFS objects not staged for commit:
> 
>         PSDs/FM_ch1-pg5.psd (LFS: 73cee30 -> File: ddb78eb)
> ```
>
> So, it appears to tracking files correctly (although, I don't know what the "%!s(MISSING)" thing is - that's new as of writing this), but when I try to commit and push changes, I get the following error.

It looks like that you have committed a file named `PSDs/FM_ch1-pg5.psd` and made additional, unstaged changes since committing. The `(%!s(MISSING))` is a bug that was fixed in https://github.com/git-lfs/git-lfs/pull/2746 and is available in the latest release of Git LFS.

> ```ShellSession
> Remote "origin" does not support the LFS locking API. Consider disabling it with:
>   $ git config 'lfs.https://gitlab.com/findmercy/ch1.git/info/lfs.locksverify' false
> Git LFS: (0 of 1 files) 0 B / 54.17 MB
> batch request: Permission denied (publickey).: exit status 255
> error: failed to push some refs to 'git@gitlab.com:findmercy/ch1.git'
> ```
>
> I assume this has something to so with the ssh key, but I'm stumped as to how to fix it. I've seen several issues similar to mine (e.g. #2635, #1068, #1726), but the solutions didn't work.

I agree that this is an issue with the SSH key and agent configuration, but the fix is dependent on your individual setup. Please make sure that you are advertising an SSH key that is accepted by GitLab and has permission to read/write the repository(s) that you're interested in.

##

As @jonmark-t notes, you must run `git push` in order to reflect changes made locally to GitLab. If the issue persists, please contact GitLab (after checking: https://docs.gitlab.com/ee/ssh/).
